dcos/marathon.py

```diff
@@ -109,12 +110,9 @@ def response_error_message(cls, status_code, reason, request_method,
             return template.format(request_method, request_url, reason, suffix)
 
         if status_code == 409:
-            _, _, path, _, _, _ = urllib.parse.urlparse(request_url)
-            for resource_name in ['app', 'group', 'pod']:
-                if '/v2/{}s'.format(resource_name) in path:
-                    break
-            else:
-                resource_name = 'resource'
+            path = urllib.parse.urlparse(request_url).path
+            path_name = (name for name in cls.RESOURCE_NAMES if name in path)
+            resource_name = next(path_name, 'resource')
```

Comment:
i would just put the last two lines in one- no need to try to find all resources in path.